### PR TITLE
Add integration tests for `DataTransformer`, remove unnecessary code and fix `max_clusters` bug

### LINKED ISSUE #312: Remove or implement CTGAN tests
Multiple [tests were never implemented](https://github.com/sdv-dev/CTGAN/blob/6373a477549200765f212cd1a9d7a856e4988fa0/tests/integration/synthesizer/test_ctgan.py#L234-L281), and are only there as scaffolding.

They should be reviewed and either deleted or implemented. CTGAN should also be investigated for any other potential useful tests, keeping in mind the cost-benefit of spending time on them.

### PR DESCRIPTION
**Pull request by fealho**
Resolve #312.

### CODE REVIEW

File: ctgan/data_transformer.py [hunk 1]
```diff
@@ -47,7 +48,9 @@ def _fit_continuous(self, data):
         """
         column_name = data.columns[0]
         gm = ClusterBasedNormalizer(
-            missing_value_generation='from_column', max_clusters=min(len(data), 10))
+            missing_value_generation='from_column',
+            max_clusters=min(len(data), self._max_clusters)
```
Comment (fealho) [inline ctgan/data_transformer.py:52]: Unrelated bug, we should pass `ClusterBasedNormalizer` `self._max_clusters` and not 10.

**Review by frances-h — APPROVED** [review]
Looks good to me 👍

File: tests/integration/test_data_transformer.py [hunk 2]
```diff
@@ -1,42 +1,146 @@
...
+        # Run
+        transformer.fit(data, ['cat'])
+        new_data = transformer.transform(data)
+        transformer.inverse_transform(new_data)
+
+        # Assert transformed numerical values are between -1 and 1
+        assert (new_data[:, 0] > -np.ones(len(new_data))).all()
+        assert (new_data[:, 0] < np.ones(len(new_data))).all()
+
+        # Assert transformed numerical values are a gaussian centerd in 0 and with std = 1/4
```
Comment (amontanez24) [inline tests/integration/test_data_transformer.py:109]: typo: centerd

File: ctgan/data_transformer.py [hunk 3]
```diff
@@ -18,8 +18,8 @@
 class DataTransformer(object):
     """Data Transformer.
 
-    Model continuous columns with a BayesianGMM and normalized to a scalar [0, 1] and a vector.
-    Discrete columns are encoded using a scikit-learn OneHotEncoder.
+    Model continuous columns with a BayesianGMM and normalizes them to a scalar between [-1, 1]
```
Comment (amontanez24) [inline ctgan/data_transformer.py:21]: minor: Either Model and normalize or models and normalizes

File: ctgan/data_transformer.py [hunk 4]
```diff
@@ -18,8 +18,8 @@
...
+    and a vector. Discrete columns are encoded using a scikit-learn's OneHotEncoder.
```
Comment (amontanez24) [inline ctgan/data_transformer.py:22]: minor: either leave as was or get rid of the 'a' before scikit

File: tests/integration/test_data_transformer.py [hunk 5]
```diff
@@ -1,42 +1,146 @@
...
+        # Run
+        transformer.fit(data, [1])
+        new_data = transformer.transform(data)
+        transformer.inverse_transform(new_data)
+
+        # Assert transformed numerical values are between -1 and 1
+        assert (new_data[:, 0] > -np.ones(len(new_data))).all()
+        assert (new_data[:, 0] < np.ones(len(new_data))).all()
+
+        # Assert transformed numerical values are a gaussian centerd in 0 and with std = 1/4
```
Comment (amontanez24) [inline tests/integration/test_data_transformer.py:138]: typo: centerd. I think every instance of that word in this file has the same problem

**Review by amontanez24 — COMMENTED** [review]
Just typos. Besides that I think this looks good!

File: ctgan/data_transformer.py (hunk 4 above)
Comment (fealho) [reply ctgan/data_transformer.py:22]: Both are wrong actually, we use RDT's OneHotEncoder...

**Review by amontanez24 — APPROVED** [review]
:shipit:

**Pull request merged**